src/rendering/framebufferrenderer.cpp

```diff
@@ -66,955 +66,963 @@ namespace {
+
+        glDeleteBuffers(1, &_vertexPositionBuffer);
+        glDeleteVertexArrays(1, &_screenQuad);
+
+        OsEng.renderEngine().raycasterManager().removeListener(*this);
+        OsEng.renderEngine().deferredcasterManager().removeListener(*this);
     }
 
-    if (_dirtyDeferredcastData) {
-        updateDeferredcastData();
+    void FramebufferRenderer::raycastersChanged(VolumeRaycaster&, bool) {
+        _dirtyRaycastData = true;
     }
 
-    // If the resolve dictionary changed (or a file changed on disk)
-    // then rebuild the resolve program.
-    if (_hdrBackGroundProgram && _hdrBackGroundProgram->isDirty()) {
-        _hdrBackGroundProgram->rebuildFromFile();
+    void FramebufferRenderer::deferredcastersChanged(Deferredcaster& deferredcaster,
+        isAttached isAttached)
+    {
+        (void)deferredcaster;
+        (void)isAttached;
+        _dirtyDeferredcastData = true;
     }
 
-    if (_resolveProgram->isDirty()) {
-        _resolveProgram->rebuildFromFile();
+    void FramebufferRenderer::update() {
+        // If the resolve dictionary changed (or a file changed on disk)
+        // then rebuild the resolve program.
+        if (_hdrBackGroundProgram && _hdrBackGroundProgram->isDirty()) {
+            _hdrBackGroundProgram->rebuildFromFile();
+        }
 
-        _uniformCache.mainColorTexture = _resolveProgram->uniformLocation(
-            "mainColorTexture"
-        );
-        _uniformCache.blackoutFactor = _resolveProgram->uniformLocation("blackoutFactor");
-        _uniformCache.nAaSamples = _resolveProgram->uniformLocation("nAaSamples");
-    }
+        if (_resolveProgram->isDirty()) {
+            _resolveProgram->rebuildFromFile();
 
-    for (auto& program : _exitPrograms) {
-        if (program.second->isDirty()) {
-            try {
-                program.second->rebuildFromFile();
-            } catch (const ghoul::RuntimeError& e) {
-                LERRORC(e.component, e.message);
+            _uniformCache.mainColorTexture = _resolveProgram->uniformLocation(
+                "mainColorTexture"
+            );
+            _uniformCache.blackoutFactor = _resolveProgram->uniformLocation("blackoutFactor");
+            _uniformCache.nAaSamples = _resolveProgram->uniformLocation("nAaSamples");
+        }
+
+        for (auto& program : _exitPrograms) {
+            if (program.second->isDirty()) {
+                try {
+                    program.second->rebuildFromFile();
+                } 
+                catch (const ghoul::RuntimeError& e) {
+                    LERRORC(e.component, e.message);
+                }
             }
         }
-    }
 
-    for (auto& program : _raycastPrograms) {
-        if (program.second->isDirty()) {
-            try {
-                program.second->rebuildFromFile();
-            } catch (const ghoul::RuntimeError& e) {
-                LERRORC(e.component, e.message);
+        for (auto& program : _raycastPrograms) {
```

Comment:
No `auto`;  use the correct type instead

Reply:
Done.